packages/custom_lint/lib/src/workspace.dart

```diff
@@ -503,24 +505,28 @@ class CustomLintProject {
   ) async {
     final directory = Directory(contextRoot.root);
 
-    final projectPubspec = await parsePubspec(directory)
-        // ignore: avoid_types_on_closure_parameters
-        .catchError((Object err, StackTrace stack) {
-      throw PubspecParseError._(
-        directory.path,
-        error: err,
-        errorStackTrace: stack,
-      );
-    });
-    final projectPackageConfig = await parsePackageConfig(directory)
-        // ignore: avoid_types_on_closure_parameters
-        .catchError((Object err, StackTrace stack) {
-      throw PackageConfigParseError._(
-        directory.path,
-        error: err,
-        errorStackTrace: stack,
-      );
-    });
+    final projectPubspec = await parsePubspec(directory).catchError(
+      // ignore: avoid_types_on_closure_parameters
+      (Object err, StackTrace stack) {
+        throw PubspecParseError._(
+          directory.path,
+          error: err,
+          errorStackTrace: stack,
+        );
+      },
+      test: (e) => e is! MissingFileError,
```

Comment:
What's the purpose of doing this?

Reply:
It forwards all exceptions that match the test, instead of catching them in catchError.

Not finding the project / pubspec.yaml / package_config.json is different than failing to parse them. Parsing can throw more errors than just the not found error. I added test for both kinds of errors.

Reply:
Updated to findProjectDirectory prior to this, so missing file errors happen before parse errors and the test here is not needed